pennylane/labs/transforms/select_pauli_rot_phase_gradient.py

```diff
@@ -0,0 +1,222 @@
+        )
+    )
+
+    for wire in phase_grad_wires:
+        ops.append(qml.ctrl(qml.X(wire), control=target_wire, control_values=[0]))
+
+    ops.append(qml.SemiAdder(angle_wires, phase_grad_wires, work_wires[: len(angle_wires) - 1]))
+
+    for wire in phase_grad_wires:
+        ops.append(qml.ctrl(qml.X(wire), control=target_wire, control_values=[0]))
+
+    ops.append(
+        qml.adjoint(qml.QROM)(
+            binary_int, control_wires, angle_wires, work_wires=work_wires[len(angle_wires) - 1 :]
+        )
+    )
+
+    return qml.prod(*ops)
+
+
+@transform
+def select_pauli_rot_phase_gradient(
+    tape: QuantumScript, angle_wires: Wires, phase_grad_wires: Wires, work_wires: Wires
+) -> tuple[QuantumScriptBatch, PostprocessingFn]:
+    r"""Quantum function transform to decompose all instances of :class:`~.SelectPauliRot` gates into additions
+    using a phase gradient resource state.
+
+    For this routine to work, the provided ``phase_grad_wires`` need to hold a phase gradient
+    state :math:`|\nabla Z\rangle = \frac{1}{\sqrt{2^n}} \sum_{m=0}^{2^n-1} e^{-2 \pi i \frac{m}{2^n}} |m\rangle`.
+    Because this state is not modified and can be re-used at a later stage, the transform does not prepare it but
+    rather assumes it has been prepared on those wires at an earlier stage.
+
+    .. figure:: ../../../_static/multiplexer_QROM.png
+                    :align: center
+                    :width: 70%
+                    :target: javascript:void(0);
+
+    The first two wires correspond to the control register, the next wire is the target qubit, followed by the register
+    equivalent to the angle wires, and finally, the register where the phase gradient is defined.
+
+    Note that this operator contains :class:`~.SemiAdder` that requires additional ``work_wires`` for the semi-in-place addition
+    :math:`\text{SemiAdder}|x\rangle_\text{ang} |y\rangle_\text{phg} = |x\rangle_\text{ang} |x + y\rangle_\text{phg}`.
+
+
+    Args:
+        tape (QNode or QuantumTape or Callable): A quantum circuit containing :class:`~.SelectPauliRot` operators.
+        angle_wires (Wires): The qubits that conditionally load the angle :math:`\phi` of
+            the :class:`~.SelectPauliRot` gate in binary as a multiple of :math:`2\pi`.
+            The length of the ``angle_wires`` implicitly determines the precision
+            with which the angle is represented.
+            E.g., :math:`(2^{-1} + 2^{-2} + 2^{-3}) 2\pi` is exactly represented by three bits as ``111``.
+        phase_grad_wires (Wires): The catalyst qubits with a phase gradient state prepared on them.
+            Needs to be at least the length of ``angle_wires`` and will only
+            use the first ``len(angle_wires)``.
+        work_wires (Wires): Additional work wires to realize the :class:`~.SemiAdder` and :class:`~.QROM`.
+            Needs to be at least ``b-1`` wires, where ``b=len(phase_grad_wires)`` is the precision of the angle :math:`\phi`.
```

Comment:
Should/Could this be `b=len(angle_wires)` instead? 

In that case, could introduce the number of angle bits once before the `Args` section and reuse it.